cmd/ipfs/main.go

```diff
@@ -509,8 +514,10 @@ func (i *cmdInvocation) setupInterruptHandler() {
 			case 0:
 				fmt.Println(shutdownMessage)
 				go func() {
+					wg.Add(1)
```

Comment:
this is racey-- the Add should go in the same goroutine as the wg gets initialized.